### DISCUSSION TITLE
How to build a library

### DISCUSSION THREAD

**Discussion by zlac** [parcel-bundler/parcel#7635] [Q&A]
Hi,

I am trying to build a js library using ES6 and parcel. I am new to bundlers and I found parcel to be easiest to setup.

I followed this guide https://parceljs.org/getting-started/library/

```
{
  "name": "my-library",
  "version": "1.0.0",
  "source": "src/index.js",
  "main": "dist/main.js",
  "scripts": {
    "watch": "parcel watch",
    "build": "parcel build"
  },
  "devDependencies": {
    "parcel": "latest"
  }
}
```

but when I try to include main.js into HTML with normal script tag like this

```
<body>
    <script src="main.js"></script>
</body>
```

I am getting error 

main.js:5 Uncaught ReferenceError: module is not defined
    at main.js:5

Also, main.js is not minified, it looks like this 

```
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "add", () => $a007c2cfd8994d7d$export$e16d8520af44a096);
function $a007c2cfd8994d7d$export$e16d8520af44a096(a, b) {
    return a + b;
}
```

What am I doing wrong?

Thanks

**Comment by windheart** [comment]
Try add type="module" to your script tag 
<script type="module" src="main.js">

**Reply by zlac** [reply]
Tried that, still the same error 

```
module is not defined
at main.js:5
```

**Reply by windheart** [reply]
Could you post your main.js here, if it is not very big?

**Reply by zlac** [reply]
I use the code from https://parceljs.org/getting-started/library/

src/index.js
```
export function add(a, b) {
  return a + b;
}
```
package.json
```
{
  "name": "my-library",
  "version": "1.0.0",
  "source": "src/index.js",
  "main": "dist/main.js",
  "devDependencies": {
    "parcel": "latest"
  }
}
```
After `npm run build` I get dist/main.js
```
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "add", () => $a007c2cfd8994d7d$export$e16d8520af44a096);
function $a007c2cfd8994d7d$export$e16d8520af44a096(a, b) {
    return a + b;
}
```

**Comment by karen-sarkisyan** [comment]
Hi,
Parcel is creating a CommonJS module in main.js, browsers don't support them.

However, modern browsers do support ES6 modules. So if you really want to import a module into your browser, you can add an option to create one in your package.json:

```diff
{
  "name": "my-library",
  "version": "1.0.0",
  "source": "src/index.js",
  "main": "dist/main.js",
+ "module": "dist/modern.main.js", // this will create an ES6 module in modern.main.js
  "devDependencies": {
    "parcel": "latest"
  }
}
```

And, as @windheart pointed out, import modern.main.js with `type="module"` attribute into your html.

**Reply by tst32** [reply]
no, the same again. no minification, all content unchanged, just vars renamed to hashnames as https://github.com/parcel-bundler/parcel/discussions/7635#discussioncomment-2088794 commented
i just want parcel to remove spaces and my comments from bundle.js

**Comment by harshsummit** [comment]
Was this issue resolved? I'm not able to import the exported functions from the bundled js file
